File: 326-power-of-three/326-power-of-three.py
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        if n <= 0:
            return False
        if n == 1:
            return True
        if n < 1:
            return self.isPowerOfThree(n / 3)
        a = n / 3
        if int(a) != a:
            return False
        return self.isPowerOfThree(int(a))

File: 326-power-of-three/326-power-of-three.cpp
class Solution {
public:
    bool isPowerOfThree(int n) {
        if (n <= 0) {
            return false;
        }
        if (n == 1) {
            return true;
        }
        if (n < 1) {
            return this->isPowerOfThree(1 / (double)n);
        }
        double a = (double)n / 3;
        if ((int) a != a) {
            return false;
        }
        return this->isPowerOfThree((int) a);
    }
};
